### ISSUE TITLE
TS2345: Argument of type 'string' is not assignable to parameter of type '`#${string}>`'.

### ISSUE THREAD

**Issue by mrjackyliang** [discordjs/discord.js#7835]
### Which package is this bug report for?

discord.js

### Issue description

Assigning a `message.channel.toString()` to an argument in another function that accepts `ChannelMention` returns a string instead of `ChannelMention`.

I've checked the types, and it seems like the `ChannelMention` type is typed correctly, but for some reason it doesn't accept it and I'm wondering why.

<img width="857" alt="Screen Shot 2022-04-24 at 7 47 39 AM" src="https://user-images.githubusercontent.com/9779803/164974911-6fdd7a49-126d-4a7d-a70b-cc0ab9fb54b4.png">

### Code sample

```typescript
client.on('messageCreate', (message) => {
  function test(abc: ChannelMention) {
    return abc;
  }

  test(message.channel.toString());
});
```

### Package version

v13.6.0

### Node.js version

v16.13.2 (TypeScript v4.6.3)

### Priority this issue should have

Medium (should be fixed soon)

### Which partials do you have configured?

No Partials

### Which gateway intents are you subscribing to?

Guilds, GuildMembers, GuildBans, GuildEmojisAndStickers, GuildInvites, GuildVoiceStates, GuildPresences, GuildMessages, GuildMessageReactions, DirectMessages

**Comment by almeidx** [comment]
`DMChannel#toString()` *should* return a `UserMention`, so you should either add `UserMention` to the union type of the `abc` parameter (*) or don't execute the function if the channel is of `DM` type.

(*) However, that won't actually fix your issue, because `PartialDMChannel#toString()` (incorrectly) returns a string

Both of those issues were fixed in #7836 (but you still have to alter your code)

**Comment by mrjackyliang** [comment]
Got it! I'll typecast it for now and look back at it later in the future.
